src/lib.rs

```diff
@@ -276,9 +321,14 @@ pub struct BindGroupLayoutEntry {
     pub ty: BindingType,
 }
 
+/// A description of a bind group layout.
 #[derive(Clone, Debug)]
 pub struct BindGroupLayoutDescriptor<'a> {
     pub bindings: &'a [BindGroupLayoutEntry],
+
+    /// An optional label to apply to the bind group layout.
+    /// This can be useful for debugging and performance analysis.
+    pub label: Option<String>,
```

Comment:
Why are the labels using owned `String` instead of `&str`? The descriptors already have lifetimes. If we're going to make these owned, we might as well consider relaxing the others (e.g. `&[T]` to `Vec<T>`).

Reply:
Some of these descriptors don't have lifetimes currently (`BufferDescriptor`, `CommandEncoderDescriptor`, `TextureDescriptor`).

We could make some `Option<String>` and others `Option<&str>` (a slight downside is that they'd be inconsistent), or we could make all `Option<&str>`  (potentially large downside that `BufferDescriptor` etc. can't be stored as easily)

Reply:
I'd say let's consistently have `Option<&str>` everywhere

Reply:
Ok :+1: will do